client/src/main/java/com/alibaba/nacos/client/naming/beat/BeatReactor.java

```diff
@@ -76,22 +76,24 @@ public String buildKey(String serviceName, String ip, int port) {
     }
 
     class BeatProcessor implements Runnable {
-
         @Override
         public void run() {
-            try {
-                for (Map.Entry<String, BeatInfo> entry : dom2Beat.entrySet()) {
-                    BeatInfo beatInfo = entry.getValue();
-                    if (beatInfo.isScheduled()) {
-                        continue;
+            while (true) {
```

Comment:
Should not use 'while(true)', consider when executorService's pool has only one thread.

Reply:
i cant get ur point  about only has one thread.